src/hotspot/os/linux/os_linux.cpp

```diff
@@ -585,22 +586,32 @@ void os::init_system_properties_values() {
     char *pslash;
     os::jvm_path(buf, bufsize);
 
-    // Found the full path to libjvm.so.
-    // Now cut the path to <java_home>/jre if we can.
+    // Found the full path to the binary. It is normally of this structure:
+    //   <jdk_path>/lib/<hotspot_variant>/libjvm.so
```

Comment:
You could change <hotspot_variant> to server I believe, since there isn't a client (I hope) version anymore.

Reply:
Okay.